src/language.ts

```diff
@@ -38,13 +38,19 @@ const standardTranslationsForLanguage = (language: string): Record<string, strin
 
 export const overrideLanguageOnContext = (manager: FlexWebChat.Manager, language: string) => {
   const appConfig = manager.configuration;
+  const [languageOnlyCode] = language.split('-');
 
   const updateConfig = {
     ...appConfig,
     context: {
       ...appConfig.context,
       language,
     },
+    componentProps: {
+      MainHeader: {
+        titleText: standardTranslations[languageOnlyCode].ChatWithUs,
+      },
+    },
```

Comment:
This makes it work, but this should **not** be here. The intention of this function is to **override language on context**.

Try moving this code (or similar) into `setNewLanguage` function, it's more appropriate there.

Reply:
Something like this:

```
const setNewLanguage = (language: string) => {
    const twilioStrings = { ...manager.strings }; // save the originals
    const defaultLanguageTranslations = standardTranslationsForLanguage(defaultLanguage);
    // eslint-disable-next-line no-shadow
    const setConfigLanguage = (language: string) => (manager.store.getState().flex.config.language = language);
    const setNewStrings = (newStrings: FlexWebChat.Strings) =>
      (manager.strings = { ...manager.strings, ...newStrings });

    if (language && language !== defaultLanguage) {
      const languageTranslations = standardTranslationsForLanguage(language);
      setConfigLanguage(language);
      setNewStrings({
        ...twilioStrings,
        ...defaultLanguageTranslations,
        ...configTranslations[defaultLanguage],
        ...languageTranslations,
        ...configTranslations[language],
      });
      setMainHeaderTitle(language); <=========== HERE
    } else {
      setConfigLanguage(defaultLanguage);
      setNewStrings({ ...twilioStrings, ...defaultLanguageTranslations, ...configTranslations[defaultLanguage] });
      setMainHeaderTitle(defaultLanguage); <=========== HERE
    }
  };
```

Reply:
@murilovmachado  - Rather than call setMainHeaderTitle() in the if/else block, why not call it outside? It still works. 

```
setMainHeaderTitle(language); <=========== HERE

if (language && language !== defaultLanguage) {
    ...
}
```

Reply:
@acedeywin  I'm not 100% confident just by looking at the if-else:
```
if (language && language !== defaultLanguage) {
  ...
} else {
  ...
}
```

It feels like language can be `null` or `undefined`.

The functions `setConfigLanguage()` and `setNewStrings()` are being repeated inside the if-else block, and probably for a reason (that I don't fully understand). I'd rather have this code repeated inside the if-else, for now.

Reply:
@murilovmachado  - Changes have been implemented.